yarpctest/recorder/recorder_test.go

```diff
@@ -126,3 +133,102 @@ func TestHash(t *testing.T) {
 	b := []byte(rgen.Atom())
 	assert.NotEqual(t, recorder.hashRequest(&r, b), referenceHash)
 }
+
+func TestOverwriteReplay(t *testing.T) {
+	dir, err := ioutil.TempDir("", "yarpcgorecorder")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer os.RemoveAll(dir) // clean up
+
+	func() {
+		// First we double check that our cache is empty.
+		recorder := NewRecorder(t, Option{
+			Mode:        Replay,
+			RecordsPath: dir,
+		})
+
+		clientDisp := yarpc.NewDispatcher(yarpc.Config{
+			Name: "client",
+			Outbounds: transport.Outbounds{
+				"server": http.NewOutbound("http://localhost:65535"),
+			},
+			Filter: recorder,
+		})
+		require.NoError(t, clientDisp.Start())
+		defer clientDisp.Stop()
+
+		client := raw.New(clientDisp.Channel("server"))
+		ctx, _ := context.WithTimeout(context.Background(), time.Second)
+
+		_, _, err := client.Call(ctx, yarpc.NewReqMeta().Procedure("hello"), []byte("Hello"))
+		require.Error(t, err)
+		assert.Contains(t, err.Error(), "no such file or directory")
+	}()
+
+	func() {
+		// Now let's record our call.
+		recorder := NewRecorder(t, Option{
+			Mode:        Overwrite,
+			RecordsPath: dir,
+		})
+
+		serverHTTP := http.NewInbound(":0")
+
+		serverDisp := yarpc.NewDispatcher(yarpc.Config{
+			Name:     "server",
+			Inbounds: []transport.Inbound{serverHTTP},
+		})
+
+		raw.Register(serverDisp, raw.Procedure("hello",
```

Comment:
You should merge the most recent dev branch. This changes to `serverDisp.Register(raw.Procedure(...))`.